src/mono/mono/metadata/appdomain.c

```diff
@@ -1256,27 +1276,135 @@ mono_runtime_install_appctx_properties (void)
 {
 	ERROR_DECL (error);
 	gpointer args [3];
+	int n_runtimeconfig_json_props = 0;
+	int n_total_props;
+	gunichar2 **total_keys;
+	gunichar2 **total_values;
+	MonoFileMap *runtimeconfig_json_map = NULL;
+	gpointer runtimeconfig_json_map_handle = NULL;
+	const char *buffer_start = runtimeconfig_json_get_buffer (runtime_config_arg, &runtimeconfig_json_map, &runtimeconfig_json_map_handle);
+	const char *buffer = buffer_start;
 
 	MonoMethod *setup = mono_class_get_method_from_name_checked (mono_class_get_appctx_class (), "Setup", 3, 0, error);
 	g_assert (setup);
 
 	// FIXME: TRUSTED_PLATFORM_ASSEMBLIES is very large
 
+	// Combine and convert properties
+	if (buffer)
+		n_runtimeconfig_json_props = mono_metadata_decode_value(buffer, &buffer);
+
+	n_total_props = n_appctx_props + n_runtimeconfig_json_props;
+	total_keys = g_new0 (gunichar2*, n_total_props);
```

Comment:
Also purely a nit but I don't like `total_keys` for the name here instead of `combined_keys` or something, `total` usually refers to a number in this sort of context.

Reply:
Fixed. Thanks!